File: src/test/java/net/dontdrinkandroot/gitki/wicket/page/user/UserListPageTest.java
package net.dontdrinkandroot.gitki.wicket.page.user;

import net.dontdrinkandroot.gitki.test.AbstractWicketTest;
import net.dontdrinkandroot.gitki.wicket.page.admin.UserListPage;
import org.junit.jupiter.api.Test;

/**
 * @author Philip Washington Sorst <philip@sorst.net>
 */
public class UserListPageTest extends AbstractWicketTest
{
    @Test
    public void testAccess()
    {
        this.assertLoginRequired(UserListPage.class);
        this.assertPageInaccessible(UserListPage.class, this.userWatcher);
        this.assertPageInaccessible(UserListPage.class, this.userCommitter);
        this.assertPageAccessible(UserListPage.class, this.userAdmin);
    }
}

File: src/test/kotlin/net/dontdrinkandroot/gitki/wicket/page/user/UserListPageTest.kt
package net.dontdrinkandroot.gitki.wicket.page.user

import net.dontdrinkandroot.gitki.test.AbstractWicketTest
import net.dontdrinkandroot.gitki.wicket.page.admin.UserListPage
import org.junit.jupiter.api.Test

class UserListPageTest : AbstractWicketTest() {

    @Test
    fun testAccess() {
        assertLoginRequired(UserListPage::class.java)
        assertPageInaccessible(UserListPage::class.java, userWatcher)
        assertPageInaccessible(UserListPage::class.java, userCommitter)
        assertPageAccessible(UserListPage::class.java, userAdmin)
    }
}